wagtail/admin/viewsets/pages.py

```diff
@@ -29,12 +32,19 @@ def get_index_view_kwargs(self, **kwargs):
         return {
             "index_url_name": self.get_url_name("index"),
             "index_results_url_name": self.get_url_name("index_results"),
+            "choose_parent_url_name": self.get_url_name("choose_parent"),
```

Comment:
Have you checked whether it's possible to just supply this as `add_url_name`, and see whether the view automatically picks it up and renders the header button accordingly?

```suggestion
            "add_url_name": self.get_url_name("choose_parent"),
```

If so, we can remove `choose_parent_url_name` and `get_choose_parent_url` from the listing view.